sync/http/handler.go

```diff
@@ -103,6 +106,19 @@ func extractFields(r *http.Request) map[string]string {
 	return f
 }
 
+func extractHeaders(r *http.Request) map[string]string {
+	h := make(map[string]string)
+
+	for name, values := range r.Header {
+		for _, value := range values {
```

Comment:
if for whatever reason the values are > 1 we will effectively overwrite the previous one.
i suggest to either use the first only (and maybe log the second one) or support the slice in the return struct.
second option and let the consumer decide...

Reply:
True, there might be an edge case where the same custom header is sent twice. 
`Header.Set()` in the `http` package sets only one key at a time 
```
// Set sets the header entries associated with key to
// the single element value. It replaces any existing
// values associated with key.
func (h MIMEHeader) Set(key, value string) {
	h[CanonicalMIMEHeaderKey(key)] = []string{value}
}
```
but from what I read [here](https://stackoverflow.com/questions/24502132/http-header-key-can-be-repeated), it can happen but it's risky since the order of the header keys is not guarranted.